### PR TITLE
Feature/gh 41 tv gamma target L1 test case

### PR DESCRIPTION
**Pull request by ashishkumarrai1998**
Updated positive and negative T.C for tvError_t GetTVGammaTarget(tvColorTemp_t colorTemp,double *x, double *y).

### CODE REVIEW

**Comment by utkarshece14** [comment]
approved

File: src/test_l1_tvSettings.c [hunk 1]
```diff
@@ -15954,32 +15956,28 @@ void test_l1_tvSettings_negative_GetTVGammaTarget (void)
     if (extendedEnumsSupported == true)
     {
         /* Step 01: Calling tvsettings GetTVGammaTarget before TvInit */
-        GetTVGammaTarget(colorTempValue, &x_Value, &y_Value);
-        UT_ASSERT_FALSE((x_Value >= 0 && x_Value <= 1.0));
+        result = GetTVGammaTarget(colorTempValue, &x_Value, &y_Value);
+        UT_ASSERT_EQUAL_FATAL(result, tvERROR_INVALID_STATE);
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:15960]: Dont use UT_ASSERT_EQUAL_FATAL instead use  UT_ASSERT_EQUAL

File: src/test_l1_tvSettings.c [hunk 2]
```diff
@@ -15954,32 +15956,28 @@ void test_l1_tvSettings_negative_GetTVGammaTarget (void)
...
     /* Step 02: Calling tvsettings initialization and expecting the API to return success */
     result = TvInit();
     UT_ASSERT_EQUAL_FATAL(result, tvERROR_NONE);
 
     /* Step 03: Calling tvsettings GetTVGammaTarget with invalid input*/
-    GetTVGammaTarget( tvColorTemp_MAX , &x_Value, &y_Value);
-    UT_ASSERT_FALSE((x_Value >= 0 && x_Value <= 1.0));
-    UT_ASSERT_FALSE((y_Value >= 0 && y_Value <= 1.0));
+    result = GetTVGammaTarget( tvColorTemp_MAX , &x_Value, &y_Value);
+    UT_ASSERT_EQUAL_FATAL(result, tvERROR_INVALID_PARAM);
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:15968]: Dont use UT_ASSERT_EQUAL_FATAL instead use UT_ASSERT_EQUAL

File: src/test_l1_tvSettings.c [hunk 3]
```diff
@@ -15954,32 +15956,28 @@ void test_l1_tvSettings_negative_GetTVGammaTarget (void)
...
 
     /* Step 04: Calling tvsettings GetTVGammaTarget with invalid input*/
-    GetTVGammaTarget( (tvColorTemp_t)-1 , &x_Value, &y_Value);
-    UT_ASSERT_FALSE((x_Value >= 0 && x_Value <= 1.0));
-    UT_ASSERT_FALSE((y_Value >= 0 && y_Value <= 1.0));
+    result = GetTVGammaTarget( (tvColorTemp_t)-1 , &x_Value, &y_Value);
+    UT_ASSERT_EQUAL_FATAL(result, tvERROR_INVALID_PARAM);
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:15972]: Dont use UT_ASSERT_EQUAL_FATAL instead use UT_ASSERT_EQUAL

File: src/test_l1_tvSettings.c [hunk 4]
```diff
@@ -15954,32 +15956,28 @@ void test_l1_tvSettings_negative_GetTVGammaTarget (void)
...
 
     /* Step 05: Calling tvsettings GetTVGammaTarget with invalid input*/
-    GetTVGammaTarget( colorTempValue, NULL, &y_Value);
-    UT_ASSERT_FALSE((x_Value >= 0 && x_Value <= 1.0));
-    UT_ASSERT_FALSE((y_Value >= 0 && y_Value <= 1.0 ));
+    result = GetTVGammaTarget( colorTempValue, NULL, &y_Value);
+    UT_ASSERT_EQUAL_FATAL(result, tvERROR_INVALID_PARAM);
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:15976]: Dont use UT_ASSERT_EQUAL_FATAL instead use UT_ASSERT_EQUAL

File: src/test_l1_tvSettings.c [hunk 5]
```diff
@@ -15954,32 +15956,28 @@ void test_l1_tvSettings_negative_GetTVGammaTarget (void)
...
 
     /* Step 06: Calling tvsettings GetTVGammaTarget with invalid input*/
-    GetTVGammaTarget( colorTempValue, &x_Value, NULL);
-    UT_ASSERT_FALSE((x_Value >= 0 && x_Value <= 1.0 ));
-    UT_ASSERT_FALSE((y_Value >= 0 && y_Value <= 1.0));
+    result = GetTVGammaTarget( colorTempValue, &x_Value, NULL);
+    UT_ASSERT_EQUAL_FATAL(result, tvERROR_INVALID_PARAM);
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:15980]: Dont use UT_ASSERT_EQUAL_FATAL instead use UT_ASSERT_EQUAL

File: src/test_l1_tvSettings.c [hunk 6]
```diff
@@ -15988,8 +15986,8 @@ void test_l1_tvSettings_negative_GetTVGammaTarget (void)
     if (extendedEnumsSupported == true)
     {
         /* Step 08: Calling tvsettings GetTVGammaTarget after TvTerm*/
-        GetTVGammaTarget(colorTempValue, &x_Value, &y_Value);
-        UT_ASSERT_FALSE((x_Value >= 0 && x_Value <= 1.0));
+        result = GetTVGammaTarget(colorTempValue, &x_Value, &y_Value);
+        UT_ASSERT_EQUAL_FATAL(result, tvERROR_INVALID_STATE);
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:15990]: Dont use UT_ASSERT_EQUAL_FATAL instead use UT_ASSERT_EQUAL

File: src/test_l2_tvSettings.c [hunk 7]
```diff
@@ -3307,7 +3307,8 @@ void test_l2_tvSettings_GetTVGammaTarget(void)
     for( colorTemp=tvColorTemp_STANDARD; colorTemp < tvColorTemp_MAX; colorTemp++ )
     {
         UT_LOG_DEBUG("Invoking GetTVGammaTarget() with colorTemp = %d", colorTemp);
-        GetTVGammaTarget(colorTemp, &x, &y);
+        status = GetTVGammaTarget(colorTemp, &x, &y);
+        UT_ASSERT_EQUAL_FATAL(status, tvERROR_NONE);
```
Comment (Anbukannadhasan) [inline src/test_l2_tvSettings.c:3311]: Dont use UT_ASSERT_EQUAL_FATAL instead use UT_ASSERT_EQUAL

File: src/test_l1_tvSettings.c [hunk 8]
```diff
@@ -15902,11 +15902,13 @@ void test_l1_tvSettings_positive_GetTVGammaTarget (void)
     {
         snprintf(keyValue, UT_KVP_MAX_ELEMENT_SIZE, "tvSettings/ColorTemperature/index/%d", i);
         colorTempValue = (tvColorTemp_t) UT_KVP_PROFILE_GET_UINT32(keyValue);
-        GetTVGammaTarget( colorTempValue, &x_Value, &y_Value);
+        result = GetTVGammaTarget( colorTempValue, &x_Value, &y_Value);
+        UT_ASSERT_EQUAL_FATAL(result, tvERROR_NONE);
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:15906]: Dont use UT_ASSERT_EQUAL_FATAL instead use UT_ASSERT_EQUAL

File: src/test_l1_tvSettings.c [hunk 9]
```diff
@@ -15902,11 +15902,13 @@ void test_l1_tvSettings_positive_GetTVGammaTarget (void)
...
         UT_ASSERT_TRUE((x_Value >= 0 && x_Value <= 1.0) );
         UT_ASSERT_TRUE((y_Value >= 0 && y_Value <= 1.0) );
 
-        GetTVGammaTarget( colorTempValue, &x_ValueRetry, &y_ValueRetry);
+        result = GetTVGammaTarget( colorTempValue, &x_ValueRetry, &y_ValueRetry);
+        UT_ASSERT_EQUAL_FATAL(result, tvERROR_NONE);
```
Comment (Anbukannadhasan) [inline src/test_l1_tvSettings.c:15911]: Dont use UT_ASSERT_EQUAL_FATAL instead use UT_ASSERT_EQUAL

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments

File: src/test_l1_tvSettings.c [hunk 10]
```diff
@@ -15953,33 +15955,29 @@ void test_l1_tvSettings_negative_GetTVGammaTarget (void)
     colorTempValue = (tvColorTemp_t)UT_KVP_PROFILE_GET_UINT32("tvSettings/ColorTemperature/index/0");
     if (extendedEnumsSupported == true)
     {
-        /* Step 01: Calling tvsettings GetTVGammaTarget before TvInit */
+        /* Step 01: Calling tvsettings GetTVGammaTarget before TvInit*/
         GetTVGammaTarget(colorTempValue, &x_Value, &y_Value);
```
[the remark this answers is not in the document]
Comment (ashishkumarrai1998) [reply src/test_l1_tvSettings.c]: addressed in commit id:- 45ba5edef84a528bb3ef3e891c40bc01d1646c0a

File: src/test_l1_tvSettings.c (hunk 1 above)
Comment (ashishkumarrai1998) [reply src/test_l1_tvSettings.c]: addressed in commit id:- 45ba5edef84a528bb3ef3e891c40bc01d1646c0a

File: src/test_l1_tvSettings.c (hunk 2 above)
Comment (ashishkumarrai1998) [reply src/test_l1_tvSettings.c]: addressed in commit id:- 45ba5edef84a528bb3ef3e891c40bc01d1646c0a

File: src/test_l1_tvSettings.c (hunk 3 above)
Comment (ashishkumarrai1998) [reply src/test_l1_tvSettings.c]: addressed in commit id:- 45ba5edef84a528bb3ef3e891c40bc01d1646c0a

File: src/test_l1_tvSettings.c (hunk 4 above)
Comment (ashishkumarrai1998) [reply src/test_l1_tvSettings.c]: addressed in commit id:- 45ba5edef84a528bb3ef3e891c40bc01d1646c0a

File: src/test_l1_tvSettings.c (hunk 5 above)
Comment (ashishkumarrai1998) [reply src/test_l1_tvSettings.c]: addressed in commit id:- 45ba5edef84a528bb3ef3e891c40bc01d1646c0a

File: src/test_l1_tvSettings.c (hunk 6 above)
Comment (ashishkumarrai1998) [reply src/test_l1_tvSettings.c]: addressed in commit id:- 45ba5edef84a528bb3ef3e891c40bc01d1646c0a

File: src/test_l2_tvSettings.c (hunk 7 above)
Comment (ashishkumarrai1998) [reply src/test_l2_tvSettings.c]: addressed in commit id:- 45ba5edef84a528bb3ef3e891c40bc01d1646c0a

File: src/test_l1_tvSettings.c (hunk 8 above)
Comment (ashishkumarrai1998) [reply src/test_l1_tvSettings.c]: addressed in commit id:- 45ba5edef84a528bb3ef3e891c40bc01d1646c0a

File: src/test_l1_tvSettings.c (hunk 9 above)
Comment (ashishkumarrai1998) [reply src/test_l1_tvSettings.c]: addressed in commit id:- 45ba5edef84a528bb3ef3e891c40bc01d1646c0a

**Review by Anbukannadhasan — APPROVED** [review]
Approved

**Review by utkarshece14 — APPROVED** [review]
Approved

**Pull request merged**